### DISCUSSION TITLE
Is it possible to externally control a Disclosure's open/close state?

### DISCUSSION THREAD

**Discussion by sebm** [twilio-labs/paste#1578] [General] [answered]
I'd like to build an experience where the Disclosure open/close state is controlled external to the Disclosure itself. Is this currently possible? If so, how?

**Comment by TheSisb** [comment, marked as answer]
Hi @sebm,

This part of our docs kind of alludes to it: https://paste.twilio.design/components/disclosure#using-state-hooks
It isn't super clear though, but yes you can tap into internal state of any of our components. Here's an example solution with Disclosure: https://codesandbox.io/s/controlled-disclosure-toggle-state-regp5?file=/pages/index.js

**Comment by SiTaggart** [comment]
We'll be adding the props to the state hook on the docs site in the next week too
